core/src/test/java/com/yahoo/imapnio/client/ImapClientRespReaderTest.java

```diff
@@ -0,0 +1,207 @@
+        final byte[] responseBytes1 = literalResponse1.getBytes(StandardCharsets.US_ASCII);
+        final ByteBuf buffer1 = Unpooled.copiedBuffer(responseBytes1, 0, responseBytes1.length);
+
+        final String literalResponse2 = ", and here u go with CRLF!\r\n";
+        final byte[] responseBytes2 = literalResponse2.getBytes(StandardCharsets.US_ASCII);
+        final ByteBuf buffer2 = Unpooled.copiedBuffer(responseBytes2, 0, responseBytes2.length);
+
+        final ByteBuf resultBuf1 = (ByteBuf) respReader.decode(null, buffer1);
+        Assert.assertNull(resultBuf1, "Should return null because not reaching a line yet");
+
+        final ByteBuf resultBuf2 = (ByteBuf) respReader.decode(null, buffer2);
+        Assert.assertNotNull(resultBuf2, "Should return null because not reaching a line yet");
+        final String result = resultBuf2.toString(StandardCharsets.US_ASCII);
+        Assert.assertEquals(result, "NO CRLF YET, and here u go with CRLF!\r\n", "decode() result mismatched.");
+    }
+
+    /**
+     * Tests input buffer has "hello 123}\r\n" where there is no left curly brace, we treat it not a literal, just return as it is.
+     *
+     * @throws Exception not for this test
+     */
+    @Test
+    public void testNoLeftCurly() throws Exception {
+        final ImapClientRespReader respReader = new ImapClientRespReader(Integer.MAX_VALUE);
+        final String literalResponse1 = "hello 123}\r\n";
+        final byte[] responseBytes1 = literalResponse1.getBytes(StandardCharsets.US_ASCII);
+        final ByteBuf buffer1 = Unpooled.copiedBuffer(responseBytes1, 0, responseBytes1.length);
+
+        final ByteBuf resultBuf1 = (ByteBuf) respReader.decode(null, buffer1);
+        Assert.assertNotNull(resultBuf1, "Should return null because not reaching a line yet");
+        final String result = resultBuf1.toString(StandardCharsets.US_ASCII);
+        Assert.assertEquals(result, literalResponse1, "decode() result mismatched.");
+    }
+
+    /**
+     * Tests input buffer has "hello {abc}\r\n" where it is not digits inside, we treat it not a literal, just return as it is.
+     *
+     * @throws Exception not for this test
+     */
+    @Test
+    public void testNotDigitsInCurly() throws Exception {
+        final ImapClientRespReader respReader = new ImapClientRespReader(Integer.MAX_VALUE);
+        final String literalResponse1 = "hello {abc}\r\n";
+        final byte[] responseBytes1 = literalResponse1.getBytes(StandardCharsets.US_ASCII);
+        final ByteBuf buffer1 = Unpooled.copiedBuffer(responseBytes1, 0, responseBytes1.length);
+
+        final ByteBuf resultBuf1 = (ByteBuf) respReader.decode(null, buffer1);
+        Assert.assertNotNull(resultBuf1, "Should return null because not reaching a line yet");
+        final String result = resultBuf1.toString(StandardCharsets.US_ASCII);
+        Assert.assertEquals(result, literalResponse1, "decode() result mismatched.");
+    }
+}
```

Comment:
add testcase where \r is passed in first call, and \n is passed in second. Also, please add a testcase where \r follows some random data, and another where \n follows some random data

Reply:
Done